core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -144,22 +147,36 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
             throttle = new IndexThrottle();
             this.searcherFactory = new SearchFactory(logger, isClosed, engineConfig);
             try {
-                writer = createWriter(openMode == EngineConfig.OpenMode.CREATE_INDEX_AND_TRANSLOG);
-                final SeqNoStats seqNoStats = loadSeqNoStatsFromCommit(writer);
+                final SeqNoStats seqNoStats;
+                final IndexCommit commitToOpen;
+                if (openMode == EngineConfig.OpenMode.CREATE_INDEX_AND_TRANSLOG) {
+                    assert engineConfig.getCommitIdToOpen() == null;
+                    commitToOpen = null;
+                    seqNoStats = new SeqNoStats(NO_OPS_PERFORMED, NO_OPS_PERFORMED, NO_OPS_PERFORMED);
+                } else if (engineConfig.getCommitIdToOpen() != null) {
+                    assert openMode == EngineConfig.OpenMode.OPEN_INDEX_CREATE_TRANSLOG;
+                    commitToOpen = findCommitById(store.directory(), engineConfig.getCommitIdToOpen());
+                    seqNoStats = Store.loadSeqNoStatsFromCommit(commitToOpen);
+                } else {
+                    // nocommit: there should be a better way
+                    commitToOpen = findCommitById(store.directory(), new Store.CommitId(store.readLastCommittedSegmentsInfo().getId()));
```

Comment:
so if you just open the IW first and get the commit from the writer and laod the stats later it should work?